src/Diagnostics.RuntimeHost/Models/InternalAPIModels.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System.Collections.Generic;
+
+namespace Diagnostics.RuntimeHost.Models
+{
+    /// <summary>
+    /// Detector Metadata.
+    /// </summary>
+    public class DetectorMetadata
```

Comment:
We already have these fields right in an existing model? Can we not reuse that or extend that

Reply:
There are slight differences between the two models. We are representing ResourceFilter as a dictionary of strings here rather than Enum. Also we don't need attributes like CodeString, PdbBytes, DllBytes for this. This model is used to send request to the Internal API.